### DISCUSSION TITLE
Does web3modal communicate with the relay server internally?

### DISCUSSION THREAD

**Discussion by dohankk** [WalletConnect/walletconnect-monorepo#3773] [Developer Support] [answered]
It says relay self hosting is not supported in v2.
Is there a direct communication from inside web3moal to the relay server?

Where in the web3modal library code can i check the relay server communication?

**Comment by glitch-txs** [comment, marked as answer]
yes, you can check the whole Web3Modal source code [here](https://github.com/WalletConnect/web3modal)
and [here](https://github.com/WalletConnect/walletconnect-monorepo)

It's abstracted as follows
Web3Modal -> Wagmi -> ethereum-provider -> universal-provider -> sign API -> core API

**Reply by dohankk** [reply]
https://github.com/WalletConnect/web3modal/blob/V3/packages/core/src/utils/CoreHelperUtil.ts#L161...L164
Is this link correct?

**Reply by glitch-txs** [reply]
nop, that's the [blockchain API](https://docs.walletconnect.com/cloud/blockchain-api)
Check the Core API in the monorepo

**Reply by dohankk** [reply]
ok thx :)
